smartsim/_core/launcher/dragon/dragonLauncher.py

```diff
@@ -395,154 +241,80 @@ def _get_managed_step_update(self, step_ids: t.List[str]) -> t.List[StepInfo]:
         :rtype: list[StepInfo]
         """
 
-        if not self.is_connected:
-            raise LauncherError("Launcher is not connected to Dragon.")
+        step_id_updates: dict[str, StepInfo] = {}
 
-        response = _assert_schema_type(
-            self._send_request(DragonUpdateStatusRequest(step_ids=step_ids)),
-            DragonUpdateStatusResponse,
-        )
-
-        # create StepInfo objects to return
-        updates: t.List[StepInfo] = []
-        # Order matters as we return an ordered list of StepInfo objects
+        dragon_step_ids: t.List[str] = []
+        slurm_step_ids: t.List[str] = []
+        pbs_step_ids: t.List[str] = []
         for step_id in step_ids:
-            if step_id not in response.statuses:
-                msg = "Missing step id update from Dragon launcher."
-                if response.error_message is not None:
-                    msg += "\nDragon backend reported following error: "
-                    msg += response.error_message
-                raise LauncherError(msg)
-
-            status, ret_codes = response.statuses[step_id]
-            if ret_codes:
-                grp_ret_code = min(ret_codes)
-                if any(ret_codes):
-                    _err_msg = (
-                        f"One or more processes failed for job {step_id}"
-                        f"Return codes were: {ret_codes}"
-                    )
-                    logger.error(_err_msg)
+            if step_id.startswith("SLURM-"):
+                slurm_step_ids.append(step_id)
+            elif step_id.startswith("PBS-"):
+                pbs_step_ids.append(step_id)
             else:
-                grp_ret_code = None
-            info = StepInfo(status, str(status), grp_ret_code)
+                dragon_step_ids.append(step_id)
+
+        if slurm_step_ids:
+            # pylint: disable-next=protected-access
+            slurm_updates = self._slurm_launcher._get_managed_step_update(
+                [
+                    DragonLauncher._unprefix_step_id(step_id)
+                    for step_id in slurm_step_ids
+                ]
+            )
+            step_id_updates.update(dict(zip(slurm_step_ids, slurm_updates)))
+
+        if pbs_step_ids:
+            # pylint: disable-next=protected-access
+            pbs_updates = self._pbs_launcher._get_managed_step_update(
```

Comment:
it feels really odd that we're not consistent in comms w/a dragon env that is started in PBS.

we should be able to raise the same events through zmq and let handlers do the necessary actions, no matter what launcher. we seem to have now coupled ourselves directly to PBS

Reply:
I think that's not the case, the PBS part is only for PBS-managed steps (PBSBatchSettings-like only).